src/test/FuzzerImpl.cpp

```diff
@@ -114,6 +115,104 @@ class FuzzTransactionFrame : public TransactionFrame
+
+    // setup the state, for this we only need to  pregenerate some accounts. For
+    // now we create 16 accounts, or enough to fill the first 4 bits such that
+    // we have a pregenerated account for the last 4 bits of the 32nd byte of a
+    // public key
+    for (int i = 0; i < 16; ++i)
+    {
+        PublicKey publicKey;
+        uint256 accountID;
+        accountID.at(31) = i;
+        publicKey.ed25519() = accountID;
+
+        // manually construct ledger entries, "creating" each account
+        LedgerEntry newAccountEntry;
+        newAccountEntry.data.type(ACCOUNT);
+        auto& newAccount = newAccountEntry.data.account();
+        newAccount.thresholds[0] = 1;
+        newAccount.accountID = publicKey;
+        newAccount.seqNum = 0;
+        // to create "interesting" balances we utilize powers of 2
+        newAccount.balance = pow(2, i);
+        ltx.create(newAccountEntry);
+
+        // select the first pregenerated account to be the hard coded source
+        // account for all transactions
+        if (i == 0)
+            mSourceAccountID = publicKey;
+    }
+
+    // commit these to the ledger so that we have a starting, persistent state
+    // to fuzz test against -- should be the only stateful/effectful action
+    ltx.commit();
+}
+
+void
+TransactionFuzzer::inject(stellar::XDRInputFileStream& in)
+{
+    // for tryRead, in case of fuzzer creating an ill-formed xdr, generate an
+    // xdr that will trigger a non-execution path so that the fuzzer realizes it
+    // has hit an uninteresting case
+    auto tryRead = [](stellar::XDRInputFileStream& in,
+                      xdr::xvector<Operation>& m) {
+        try
+        {
+            return in.readOne(m);
+        }
+        catch (...)
+        {
+            m.clear(); // we ignore transactions with 0 operations
+            return true;
+        }
+    };
+
+    xdr::xvector<Operation> ops;
+    while (tryRead(in, ops))
+    {
+        // direct the fuzzer such that it will "learn" that five transactions
+        // is the maximum since nothing over five transactions will ever
+        // result in newly discovered execution paths
+        if (ops.size() == 0 || ops.size() > 5)
+            return;
+
+        // construct transaction
+        auto txFramePtr = ::FuzzTransactionFrame::create(mSourceAccountID, ops,
+                                                         mApp->getNetworkID());
+
+        try
+        {
+            // attempt to apply transaction
+            LedgerTxn ltx(mApp->getLedgerTxnRoot());
+            txFramePtr->attemptApplication(*mApp, ltx);
+
+            // we need an extra db cache clear here since the db prepared
+            // statement cache may persist between tryRead loops
+            mApp->getDatabase().clearPreparedStatementCache();
+        }
+        catch (InvariantDoesNotHold const&)
+        {
+            throw;
+        }
+        catch (TransactionResultCode const& err)
```

Comment:
I don't think we ever actually throw a `TransactionResultCode`. How is this block reached? I think to catch internal errors correctly, you need to interpret any exception other than `InvariantDoesNotHold` as an internal error. Based on the comment in the next block, this might be a non-trivial change requiring some validity checks to filter out other error sources.

Reply:
Ok, so you are correct, TransactionResultCode is never thrown. It was thrown in the original version of the PR, but is no longer, so that has been removed. 

Also, I figured out that the **other** exceptions, such as the ones I note in the comment are due to me removing a `checkValid` call. I originally removed this because I though it just checked signatures and such -- however it checks if operations are well-formed as well, and as far as I can tell, the other exceptions being caught were all mal-formed XDR exceptions such as `terminate called after throwing an instance of 'xdr::xdr_wrong_union'`.

My solution for this is re-including the `checkValid` and if it fails for any operation, returning early, and not *"applying the transaction"*.